lib/utilities/package-cache.js

```diff
@@ -0,0 +1,480 @@
+      if (!fs.existsSync(directory)) {
+        this._conf.delete(label);
+      }
+    }
+  },
+
+  /**
+   * The `_readManifest` method reads the on-disk manifest for the current
+   * cache and returns its value.
+   *
+   * @method _readManifest
+   * @param {String} label The label for the cache.
+   * @param {String} type The type of package cache.
+   * @return {String} The manifest file contents on disk.
+   */
+  _readManifest: function(label, type) {
+    var readManifestDir = this.dirs[label];
+
+    if (!readManifestDir) { return null; }
+
+    var inputPath = path.join(readManifestDir, translate(type, 'manifest'));
+
+    var result = null;
+    try {
+      result = fs.readFileSync(inputPath, 'utf8');
+    } catch(error) {
+      // Swallow non-exceptional errors.
+      if (error.code !== 'ENOENT') {
+        throw error;
+      }
+    }
+    return result;
+  },
+
+  /**
+   * The `_writeManifest` method generates the on-disk folder for the package cache
+   * and saves the manifest into it. If it is a yarn package cache it will remove
+   * the existing lock file.
+   *
+   * @method _writeManifest
+   * @param {String} label The label for the cache.
+   * @param {String} type The type of package cache.
+   * @param {String} manifest The contents of the manifest file to write to disk.
+   */
+  _writeManifest: function(label, type, manifest) {
+    var outputDir = quickTemp.makeOrReuse(this.dirs, label);
+    this._conf.set(label, outputDir);
+
+    var outputFile = path.join(outputDir, translate(type, 'manifest'));
+    fs.writeFileSync(outputFile, manifest);
+
+    // Remove any existing yarn.lock file so that it doesn't try to incorrectly use it as a base.
+    if (type === 'yarn') {
+      try {
+        fs.unlinkSync(path.join(outputDir, 'yarn.lock'));
+      } catch (error) {
+        // Catch unexceptional error but rethrow if something is truly wrong.
+        if (error.code !== 'ENOENT') {
+          throw error;
+        }
+      }
+    }
+  },
+
+  /**
+   * The `_checkManifest` method compares the desired manifest to that which
+   * exists in the cache.
+   *
+   * @method _checkManifest
+   * @param {String} label The label for the cache.
+   * @param {String} type The type of package cache.
+   * @param {String} manifest The contents of the manifest file to compare to cache.
+   * @return {Boolean} `true` if identical.
+   */
+  _checkManifest: function(label, type, manifest) {
+    var cachedManifest = this._readManifest(label, type);
+
+    if (cachedManifest === null) { return false; }
+
+    var parsedCached = JSON.parse(cachedManifest);
+    var parsedNew = JSON.parse(manifest);
+
+    // Only inspect the keys we care about.
+    var keys = ['dependencies', 'devDependencies', 'peerDependencies', 'optionalDependencies'];
+
+    var key;
+    for (var i = 0; i < keys.length; i++) {
+      key = keys[i];
+      if (JSON.stringify(parsedCached[key]) !== JSON.stringify(parsedNew[key])) {
+        return false;
```

Comment:
My tests for the fixtures were unnecessarily invalidating because of this and it was really frustrating. So, solved. 😄